### PR TITLE
Initial login modal work

### PR DESCRIPTION
**Pull request by gjvera**
image (3)
image (4)

### CODE REVIEW

File: src/main/webapp/result-forms/result-forms.stories.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {
 } from '@connexta/ace/@storybook/addon-knobs'
 import { action } from '@connexta/ace/@storybook/addon-actions'
 
-const stories = storiesOf('ResultFroms', module)
+const stories = storiesOf('ResultForms', module)
```
Comment (djblue) [inline src/main/webapp/result-forms/result-forms.stories.js]: Thanks for the fix :+1:

File: src/main/webapp/login/login.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        flexDirection: 'column',
+        height: '100%',
+        maxHeight: '100%',
+        padding: 20,
+        boxSizing: 'border-box',
+        ...props.style,
+      }}
+    >
+      <FormLabel style={{ padding: 10 }}>{props.label}</FormLabel>
+      <TextField
```
Comment (djblue) [inline src/main/webapp/login/login.js]: How are you getting the values form the textfields for the login request?

File: src/main/webapp/intrigue-api/graphql.js [hunk 3]
```diff
@@ -381,6 +381,24 @@ const Query = {
...
+const btoa = arg => {
+  if (typeof window !== 'undefined') {
+    return window.btoa(arg)
+  }
+  return Buffer.from(arg).toString('base64')
+}
+
+const logIn = async (parent, args, { fetch }) => {
+  const { username, password } = args
+  const authToken = btoa(`${username}:${password}`)
```
Comment (djblue) [inline src/main/webapp/intrigue-api/graphql.js]: Don't you need to include the method of auth as well? `Authorization: Basic ${auth}`

File: src/main/webapp/login/login.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        onBlur={handleChange('username')}
+      />
+      <FormControl
+        fullWidth
+        required
+        variant="outlined"
+        style={{ marginBottom: 20 }}
+      >
+        <InputLabel>Password</InputLabel>
+        <OutlinedInput
```
Comment (djblue) [inline src/main/webapp/login/login.js]: Why is this not a TextField component?

File: src/main/webapp/login/login.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    >
+      <FormLabel style={{ padding: 10 }}>{props.label}</FormLabel>
+      <TextField
+        autoFocus
+        required
+        fullWidth
+        variant="outlined"
+        label="Username"
+        style={{ marginBottom: 20 }}
+        onBlur={handleChange('username')}
```
Comment (djblue) [inline src/main/webapp/login/login.js]: I think you should use value and onChange for input fields.

File: src/main/webapp/login/login.js (hunk 4 above)
Comment (gjvera) [reply src/main/webapp/login/login.js]: The text field component doesn't support the `EndAdornment` Prop which I added to let the user toggle the visibility of their password in the modal

File: src/main/webapp/intrigue-api/context.js [hunk 6]
```diff
@@ -1,18 +1,22 @@
...
-    return fetch(url, {
+  const universalFetch = async (url, opts = {}) => {
+    const res = await fetch(url, {
       ...opts,
       headers: {
         ...opts.headers,
         authorization,
       },
     })
+    if (res.status === 401) {
```
Comment (gjvera) [inline src/main/webapp/intrigue-api/context.js]: Going to add more functionality for different kinds of errors. Authentication is just the first one.

File: src/main/webapp/intrigue-api/graphql.js [hunk 7]
```diff
@@ -30,24 +45,55 @@ const createServerApollo = (...args) => {
...
   })
 }
 
+const clientErrorLink = onError(
+  ({ graphQLErrors, networkError, operation, forward }) => {
+    if (graphQLErrors) {
+      for (let err of graphQLErrors) {
+        if (err.extensions.code === 'UNAUTHENTICATED') {
+          //open modal and get credientials
+          const oldHeaders = operation.getContext().headers
```
Comment (gjvera) [inline src/main/webapp/intrigue-api/graphql.js]: Once the credentials from the modal are passed, need to find the best way to permanently set the context to use that header.

File: src/main/webapp/login/login.js [hunk 8]
```diff
@@ -75,11 +78,18 @@ export const LogIn = props => {
...
-        onClick={() => {
+        disabled={values.buttonDisabled ? true : false}
+        onClick={async () => {
+          setValues({ ...values, buttonDisabled: true })
           //TO:DO parse cookie from gql query and set it
-          logIn({
+          const { data } = await logIn({
             variables: { username: values.username, password: values.password },
           })
+          const origCookie = data.logIn
```
Comment (gjvera) [inline src/main/webapp/login/login.js]: component shouldn't be doing this logic. move outside

File: src/main/webapp/login/login.js [hunk 9]
```diff
@@ -75,11 +78,18 @@ export const LogIn = props => {
...
+          let parsedCookie = origCookie.split(';')
+          parsedCookie = parsedCookie[0].split('=')
+          cookies.set(parsedCookie[0], parsedCookie[1], { path: '/' })
+          console.log(cookies.get(parsedCookie[0]))
```
Comment (gjvera) [inline src/main/webapp/login/login.js]: remove

**Comment by gjvera** [comment]
Gonna clean this up and reopen

**Pull request closed without merging**